### DISCUSSION TITLE
traditional machine learning model

### DISCUSSION THREAD

**Discussion by wangjiawen2013** [tracel-ai/burn#1683] [Q&A]
Hi,
It is said that "Burn aims to build a full-fledged machine learning stack in Rust" and the following example are presented (https://medium.com/@athan.seal/candle-vs-burn-comparing-rust-machine-learning-frameworks-4dbd59c332a1):
```
use burn::prelude::*;

fn main() {
    let dataset = Dataset::from_csv("data.csv")?;
    let model = LogisticRegression::new(dataset.num_features());

    let optimizer = GradientDescent::new(0.01);
    let mut history = optimizer.fit(&dataset, &model, 100)?;

    println!("Training loss: {}", history.losses().last().unwrap());
}
```
Where can I find more examples on traditional machine learning such as logisticregression, svm, dimension reduction, naive bayes and so on ?
I didn't find LogisticRegression on burn's documentation. Is the above comment on burn and candle true ?

**Comment by laggui** [comment]
Actually, Burn is not meant to be a "full-fledged machine learning stack". Burn is a *deep learning* framework.

We do not have traditional machine learning methods such as svm, naive bayes, etc.

Not sure if the linked post included the code to their example, but the closest thing we have is a [simple regression example](https://github.com/tracel-ai/burn/tree/main/examples/simple-regression).

If you're looking for traditional ML methods in Rust, take a look at [Are We Learning Yet](https://www.arewelearningyet.com/). I just stumbled upon this recently, it could be a helpful resource for you.

**Comment by nathanielsimard** [comment]
The blog post seems low quality and probably AI generated. Just by reading our website or the GitHub readme, you can see that Burn puts a lot of thought into performance, not just flexibility.
